python/tests/__init__.py

```diff
@@ -110,6 +110,30 @@ def _levelorder_nodes(self, u, l, level):
         for c in self.children(u):
             self._levelorder_nodes(c, l, level + 1)
 
+    def _minlex_postorder_nodes(self, u, l):
+        l.extend(self._minlex_postorder_nodes_helper(u)[1])
+
+    def _minlex_postorder_nodes_helper(self, u):
+        """
+        For a given input ID u, this function returns a tuple whose first value
+        is the minimum leaf node ID under node u, and whose second value is
+        a list containing the minlex postorder for the subtree rooted at node u.
+        The first value is needed for sorting, and the second value is what
+        finally gets returned.
+        """
+        children = self.children(u)
+        if len(children) > 0:
+            children_return = [self._minlex_postorder_nodes_helper(c) for c in children]
+            # sorts by first value, which is the minimum leaf node ID
+            children_return.sort()
+            minlex_postorder = []
+            for _, child_minlex_postorder in children_return:
+                minlex_postorder.extend(child_minlex_postorder)
+            minlex_postorder.extend([u])
+            return (children_return[0][0], minlex_postorder)
```

Comment:
Note that we can't use sorted(children_return) above because we need to get the first element here as well